pkg/asset/installconfig/azure/client.go

```diff
@@ -328,3 +329,28 @@ func (c *Client) AreMarketplaceImageTermsAccepted(ctx context.Context, publisher
 
 	return terms.AgreementProperties.Accepted != nil && *terms.AgreementProperties.Accepted, nil
 }
+
+// GetAvailabilityZones retrieves a list of availability zones for the given region, and instance type.
+func (c *Client) GetAvailabilityZones(ctx context.Context, instanceType string, region string) ([]string, error) {
+	client := azenc.NewResourceSkusClientWithBaseURI(c.ssn.Environment.ResourceManagerEndpoint, c.ssn.Credentials.SubscriptionID)
+	var zones []string
+	// Only supported filter atm is `location`
+	filter := fmt.Sprintf("location eq '%s'", region)
+	for res, err := client.List(ctx, filter); res.NotDone(); err = res.NextWithContext(ctx) {
+		if err != nil {
+			return zones, err
+		}
+
+		for _, resSku := range res.Values() {
+			if strings.EqualFold(to.String(resSku.Name), instanceType) {
+				for _, locationInfo := range *resSku.LocationInfo {
+					if strings.EqualFold(to.String(locationInfo.Location), region) {
```

Comment:
just curious: now that we have the `filter`, is this still needed, or is this just defensive programming? I am fine with leaving it in, defensively (perhaps leave a comment), but it seems like it would not be needed?

Reply:
I don't think it's needed any more. Maybe we can remove it and wait for a round of CI to check it doesn't impact anything?